### PR TITLE
fix: add Bazel option to force creating bazel-bin symlink

### LINKED ISSUE #5185: bazel-bot must explicitly request bazel symlinks to be created
Problem: https://github.com/googleapis/googleapis/commit/43f02766add4ac5e2e6557a9d2600f2101654dff broke Bazel bot. It was reverted, but we (@noahdietz and I) now want to fix Bazel bot to try that change again.

Background: when Bazel build completes, Bazel will create several symlinks (`bazel-bin`, `bazel-out`, etc.) in the workspace root. These symlinks are only created for convenience ([doc](https://bazel.build/remote/output-directories#:~:text=The%20symlinks%20for%20%22bazel%2D%3C,itself%20does%20not%20use%20them.)). Unfortunately, for most users, it's an _inconvenience_, because these symlinks often get into CLs (since it all happens inside the workspace).

Bazel has an option to disable symlink creation (see the commit above). But turns out Bazel bot [expects](https://github.com/googleapis/repo-automation-bots/blob/222c462ffa607d2042a84f40cf82acc4d5c527a0/packages/bazel-bot/docker-image/generate-googleapis-gen.sh#L122) `bazel-bin` symlink to be created.

There are two options to fix it:

(1) what I suggest in #5183: Bazel bot will ask Bazel to create symlinks even if it's disabled in `.bazelrc`;
(2) alternative option is to stop relying on `bazel-bin` to find the tarballs, but use `bazel info bazel-bin` to find the real location:

```
$ bazel info bazel-bin
Starting local Bazel server and connecting to it...
/usr/local/google/home/fenster/.cache/bazel/_bazel_fenster/56931f8ec0ff6fd79dcc1b03ef38d04e/execroot/com_google_googleapis/bazel-out/k8-fastbuild/bin
```

I can do either (1) or (2), your choice.

### PR DESCRIPTION
**Pull request by alexander-fenster**
Fixes #5185.

These `bazel-*` symlinks add a lot of toil for human users, and we are thinking of disabling them in .bazelrc. Since this code depends on them being created, let's have them enabled explicitly in the script.

`BAZEL_FLAGS` is not the right place for it, because this is not something that is supposed to be configured, but rather something that is supposed to be hardcoded.

### CODE REVIEW

**Comment by alexander-fenster** [comment]
CI fails with

```
data: {
      message: "API rate limit exceeded for 40.84.171.158. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)",
      documentation_url: 'https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting'
    },
```

so I guess it will maybe pass eventually.

**Comment by bcoe** [comment]
Adding @SurferJeffAtGoogle who is a better reviewer for bazel related issues.

**Comment by SurferJeffAtGoogle** [comment]
I don't understand the problem from the PR description.  Could you open an issue and describe the problem in more detail?

**Comment by alexander-fenster** [comment]
@SurferJeffAtGoogle I filed #5185.

File: packages/bazel-bot/docker-image/generate-googleapis-gen.sh [hunk 1]
```diff
@@ -103,7 +103,8 @@ for (( idx=${#ungenerated_shas[@]}-1 ; idx>=0 ; idx-- )) ; do
     # APIs from being updated.
     set +e
     # Invoke bazel build.
-    (cd "$GOOGLEAPIS" && bazelisk build $BAZEL_FLAGS -k $targets)
+    # --experimental_convenience_symlinks=normal is to make sure bazel-bin symlink is created
+    (cd "$GOOGLEAPIS" && bazelisk build --experimental_convenience_symlinks=normal $BAZEL_FLAGS -k $targets)
```
Comment (SurferJeffAtGoogle) [inline packages/bazel-bot/docker-image/generate-googleapis-gen.sh:107]: The `experimental_` prefix on this flag name tells me not to depend on it for the long term.  I expect it to be dropped or renamed within a year or two.

Therefore, I'd prefer invoking `BAZEL_BIN=$(bazel info bazel-bin)` as you described in the issue.

Comment (alexander-fenster) [reply packages/bazel-bot/docker-image/generate-googleapis-gen.sh]: Done, PTAL.  Also, how do I test it properly?

**Comment by alexander-fenster** [comment]
@SurferJeffAtGoogle The `finale` job keeps failing in this PR, do you have any idea what's going on here?

**Review by SurferJeffAtGoogle — APPROVED** [review]
New code looks good.  To test, run this script:
https://github.com/googleapis/repo-automation-bots/blob/main/packages/bazel-bot/test-generate-googleapis-gen.sh

**Comment by SurferJeffAtGoogle** [comment]
> @SurferJeffAtGoogle The `finale` job keeps failing in this PR, do you have any idea what's going on here?

finale passed.  Please feel free to merge after running test.

**Comment by alexander-fenster** [comment]
@SurferJeffAtGoogle `finale` still not passing :(

**Comment by alexander-fenster** [comment]
@SurferJeffAtGoogle ping on this one once more :)

**Comment by SurferJeffAtGoogle** [comment]
@alexander-fenster All tests passed.  Please merge at your convenience.

**Pull request merged**